api-tests/specs/role-service/roleservice.js

```diff
@@ -0,0 +1,41 @@
+const requests = require('../../data/role-service/roleservice')
+const conf = require('../../config/conf')
+
+beforeEach(function () {
+    baseRequest = request(conf.BASE_URL)
+})
+
+describe('As a user I want to create roles so that they can be assigned to persons', function () {
+    it('I am able to add a role to the trial site successfully', async () => {
+        const response = await baseRequest.post('/roles').send(requests.validRole);
+        expect(response.text).to.contain("id")
+        expect(response.status).to.equal(HttpStatus.OK)
+    });
+
+    it('User is prevented from saving the new role if the name is already used for an existing role', async () => {
+        const roleName = requests.validRole
+        let response = await baseRequest.post('/roles').send(roleName);
+        response = await baseRequest.post('/roles').send(roleName);
+        expect(response.status).to.equal(409)
```

Comment:
Is it better to use HttpStatus.CONFLICT rather than 409?    It is the same thing of course but might be more understandable for future developers?

Reply:
that's a very valid point, but the response status is sent back as either 404 or 409 but no error text attached to the status as below
I can use an additional assertion like,  expect(response.body).to.contain(Conflict) if that helps

`{"timestamp":"2021-01-21T11:03:54.070+00:00","status":409,"error":"Conflict","message":"Duplicate role ID","path":"/roles"}`

Reply:
Thanks Kate, I have found a way to handle it :)